### PR TITLE
HIVE-25716 Fix of flaky test TestCompactionMetrics#testOldestReadyForCleaningAge

### PR DESCRIPTION
**Pull request by vcsomor**
Fix of flaky test TestCompactionMetrics#testOldestReadyForCleaningAge

### CODE REVIEW

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java [hunk 1]
```diff
@@ -178,41 +178,44 @@ public void testInitiatorPerfMetricsDisabled() throws Exception {
...
-    burnThroughTransactions("default", "young_rfc", 25);
-    rqst = new CompactionRequest("default", "young_rfc", CompactionType.MINOR);
-    rqst.setPartitionname("ds=part");
-    txnHandler.compact(rqst);
-    startWorker();
+    burnThroughTransactions(DB_NAME, YOUNG_TABLE_NAME, 25);
+    doCompaction(DB_NAME, YOUNG_TABLE_NAME, YOUNG_PARTITION_NAME, CompactionType.MINOR);
+    long youngTableClosestCqCommitTime = System.currentTimeMillis();
+
+    Thread.sleep(1500);
```
Comment (deniskuzZ) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java:208]: what's the purpose of 1.5 sec sleep?

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java [hunk 2]
```diff
@@ -178,41 +178,42 @@ public void testInitiatorPerfMetricsDisabled() throws Exception {
...
+    long nowClosestToAcidMetricServiceEnd = System.currentTimeMillis();
 
-    long threshold = 1000;
-    Assert.assertTrue(Metrics.getOrCreateGauge(MetricsConstants.OLDEST_READY_FOR_CLEANING_AGE).intValue() <= oldDiff + threshold);
-    Assert.assertTrue(Metrics.getOrCreateGauge(MetricsConstants.OLDEST_READY_FOR_CLEANING_AGE).intValue() >= youngDiff);
+    long oldAgeInSeconds = (nowClosestToAcidMetricServiceEnd - oldTableClosestCqCommitTime) / 1000;
+    long youngAgeInSeconds = (nowClosestToAcidMetricServiceEnd - youngTableClosestCqCommitTime) / 1000;
+
+    int gaugeValue = Metrics.getOrCreateGauge(MetricsConstants.OLDEST_READY_FOR_CLEANING_AGE).intValue();
+    Assert.assertTrue(gaugeValue <= oldAgeInSeconds);
```
Comment (deniskuzZ) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestCompactionMetrics.java:215]: why not check that the gauge value is in some range of `oldAgeInSeconds` (+- 10ms)? Not sure why youngAgeInSeconds is even relevant here.

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/CompactorTest.java [hunk 3]
```diff
@@ -292,22 +292,24 @@ protected void burnThroughTransactions(String dbName, String tblName, int num, S
...
-      assert(awiResp.getTxnToWriteIds().get(i++).getTxnId() == tid);
-      if(lockReq != null) {
+      assert (awiResp.getTxnToWriteIds().get(i).getTxnId() == tid);
+      ++i;
+      if (lockReq != null) {
         lockReq.setTxnid(tid);
         txnHandler.lock(lockReq);
       }
-      if (aborted != null && aborted.contains(tid)) {
+      if ((aborted != null) && aborted.contains(tid)) {
```
Comment (deniskuzZ) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/CompactorTest.java:310]: unnecessary brackets

File: ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/CompactorTest.java [hunk 4]
```diff
@@ -292,22 +292,24 @@ protected void burnThroughTransactions(String dbName, String tblName, int num, S
...
         txnHandler.abortTxn(new AbortTxnRequest(tid));
-      } else if (open == null || (open != null && !open.contains(tid))) {
+      } else if ((open == null) || !open.contains(tid)) {
```
Comment (deniskuzZ) [inline ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/CompactorTest.java:312]: unnecessary brackets

**Review by deniskuzZ — APPROVED** [review]

**Comment by vcsomor** [comment]
Root cause was:
```java
Assert.assertTrue(Metrics.getOrCreateGauge(MetricsConstants.OLDEST_READY_FOR_CLEANING_AGE).intValue() >= youngDiff);
```
Output:
```
Now: 1638863282765
Old start: 1638863245187
Young start: 1638863263762
Old diff: 37 (diff 37578)
Young diff: 19 (19003)
Age: 18 (which is calculated and rounded inside the runAcidMetricsService)
```

**Review by asinkovits — APPROVED** [review]
LGTM +1

**Pull request merged**